cce/cce.go

```diff
@@ -31,55 +30,51 @@ func New(sheetsServerString, baseDir string) *Handler {
 		sheetsServerString: sheetsServerString,
 		baseDir:            baseDir,
 		status:             status.Idle,
-		err:                "",
 	}
 }
 
 // Get returns current state and last error
 func (h *Handler) Get(c echo.Context) error {
-	payload := make(map[string]interface{})
-	payload["errorMessage"] = h.err
-	payload["status"] = h.status
-	return c.JSON(http.StatusOK, payload)
+	return c.JSON(http.StatusOK, map[string]interface{}{
+		"errorMessage": h.err,
+		"status":       h.status,
+	})
+}
+
+func (h *Handler) post(c echo.Context) {
+	in := postRequest{}
+	if err := c.Bind(&in); err != nil {
+		handleError(fmt.Sprintf("o corpo da requisicão enviado é inválido: %q", err), h)
+		return
+	}
+	downloadURL := fmt.Sprintf(h.sheetsServerString, in.Year)
+	zipFileName := fmt.Sprintf("cce_sheets_%d.zip", in.Year)
+	f, err := os.Create(zipFileName)
+	if err != nil {
+		handleError(fmt.Sprintf("ocorreu uma falha durante a criação dos arquivos zip com nome %s, erro: %q", zipFileName, err), h)
+		return
+	}
+	err = donwloadFile(downloadURL, f)
+	if err != nil {
+		handleError(fmt.Sprintf("ocorreu uma falha ao fazer o download dos arquivos csv da legislatura %d pelo link %s, errro: %q", in.Year, downloadURL, err), h)
+		return
+	}
+	h.status = status.Processing
 }
 
 // Post implements a post request for this handler
 func (h *Handler) Post(c echo.Context) error {
 	if h.status != status.Idle {
-		return c.JSON(http.StatusServiceUnavailable, "sistema está processando dados")
+		return c.String(http.StatusServiceUnavailable, "sistema está processando dados")
 	}
 	h.status = status.Collecting
```

Comment:
Acho que essa mudança de status deve acontecer somente depois da validação da entrada. Ou seja, na linha 49.

Inclusive, poderia ter um teste que confira que se a entrada for inválida o status seja idle? Isso é bem importante, pois pode travar o status do sistema.